mmengine/structures/instance_data.py

```diff
@@ -165,9 +177,8 @@ def __getitem__(self, item: IndexType) -> 'InstanceData':
             # More details in https://github.com/numpy/numpy/issues/9464
             item = item.astype(np.int64) if item.dtype == np.int32 else item
             item = torch.from_numpy(item)
-        assert isinstance(
-            item, (str, slice, int, torch.LongTensor, torch.cuda.LongTensor,
-                   torch.BoolTensor, torch.cuda.BoolTensor))
+        assert isinstance(item, (str, slice, int, LongTypeTensor.__args__,
+                                 BoolTypeTensor.__args__))
```

Comment:
```suggestion
        assert isinstance(item, IndexType.__args__)
```

Reply:
And move this line to Line 171.

```python
assert isinstance(item, IndexType.__args__)
if isinstance(item, list):
    item = np.array(item)
if isinstance(item, np.ndarray):
    # The default int type of numpy is platform dependent, int32 for
    # windows and int64 for linux. `torch.Tensor` requires the index
    # should be int64, therefore we simply convert it to int64 here.
    # More details in https://github.com/numpy/numpy/issues/9464
    item = item.astype(np.int64) if item.dtype == np.int32 else item
    item = torch.from_numpy(item)
```

Reply:
Thanks for the suggestion, already fixed.